Build failure with 2.40.4 on HPPA (test_mkfds_ppoll.c:68:26: error: unknown type name ‘sigset_t’; did you mean ‘size_t’?)

On HPPA, when building util-linux-2.40.4's tests, I hit:
```
hppa2.0-unknown-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I/var/tmp/portage/sys-apps/util-linux-2.40.4/work/util-linux-2.40.4  -include config.h -I/var/tmp/portage/sys-apps/util-linux-2.40.4/work/util-linux-2.40.4/include -DLOCALEDIR=\"/usr/share/locale\" -D_PATH_RUNSTATEDIR=\"/run\" -D_PATH_LOCALSTATEDIR=\"/var\" -D_PATH_SYSCONFSTATICDIR=\"/usr/lib\"    -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -fsigned-char -fno-common -Wall -Wextra -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -Werror=sequence-point -O2 -pipe -march=2.0 -fdiagnostics-color=always -frecord-gcc-switches -c -o tests/helpers/test_mkfds_ppoll.o /var/tmp/portage/sys-apps/util-linux-2.40.4/work/util-linux-2.40.4/tests/helpers/test_mkfds_ppoll.c
/var/tmp/portage/sys-apps/util-linux-2.40.4/work/util-linux-2.40.4/tests/helpers/test_mkfds_ppoll.c:68:26: error: unknown type name ‘sigset_t’; did you mean ‘size_t’?
   68 | static void clear_sigset(sigset_t *sigset)
      |                          ^~~~~~~~
      |                          size_t
In file included from /var/tmp/portage/sys-apps/util-linux-2.40.4/work/util-linux-2.40.4/tests/helpers/test_mkfds_ppoll.c:38:
/var/tmp/portage/sys-apps/util-linux-2.40.4/work/util-linux-2.40.4/tests/helpers/test_mkfds_ppoll.c: In function ‘wait_event_ppoll’:
/var/tmp/portage/sys-apps/util-linux-2.40.4/work/util-linux-2.40.4/tests/helpers/test_mkfds_ppoll.c:76:23: error: unknown type name ‘sigset_t’; did you mean ‘size_t’?
   76 |                       sigset_t sigset;,
      |                       ^~~~~~~~
/var/tmp/portage/sys-apps/util-linux-2.40.4/work/util-linux-2.40.4/tests/helpers/test_mkfds.h:47:17: note: in definition of macro ‘DEFUN_WAIT_EVENT_POLL’
   47 |                 XDECLS                                                  \
      |                 ^~~~~~
/var/tmp/portage/sys-apps/util-linux-2.40.4/work/util-linux-2.40.4/tests/helpers/test_mkfds_ppoll.c:77:23: error: implicit declaration of function ‘clear_sigset’ [-Wimplicit-function-declaration]
   77 |                       clear_sigset(&sigset);,
      |                       ^~~~~~~~~~~~
/var/tmp/portage/sys-apps/util-linux-2.40.4/work/util-linux-2.40.4/tests/helpers/test_mkfds.h:73:17: note: in definition of macro ‘DEFUN_WAIT_EVENT_POLL’
   73 |                 SETUP_SIG_HANDLER                                       \
      |                 ^~~~~~~~~~~~~~~~~
/var/tmp/portage/sys-apps/util-linux-2.40.4/work/util-linux-2.40.4/tests/helpers/test_mkfds_ppoll.c:77:23: warning: nested extern declaration of ‘clear_sigset’ [-Wnested-externs]
   77 |                       clear_sigset(&sigset);,
      |                       ^~~~~~~~~~~~
/var/tmp/portage/sys-apps/util-linux-2.40.4/work/util-linux-2.40.4/tests/helpers/test_mkfds.h:73:17: note: in definition of macro ‘DEFUN_WAIT_EVENT_POLL’
   73 |                 SETUP_SIG_HANDLER                                       \
      |                 ^~~~~~~~~~~~~~~~~
make[3]: *** [Makefile:10910: tests/helpers/test_mkfds_ppoll.o] Error 1
```

This is with linux-headers-6.6 and glibc-2.40.

--- a/configure.ac
+++ b/configure.ac
@@ -987,6 +987,15 @@ AC_CHECK_TYPES([sighandler_t], [], [], [[
 #include <signal.h>
 ]])
 
+# checking sigset_t defined in the kernel header in
+# asm/signal.h. tests/helpers/test_mkfds_ppoll.c expects the RAW sigset_t
+# is defined in the header file.
+# However, a slightly older arch/parisc doesn't export the type.
+# See https://web.git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=487fa28fa8b60417642ac58e8beda6e2509d18f9
+AC_CHECK_TYPES([sigset_t], [], [], [[
+#include <asm/signal.h>
+]])
+
 AC_CHECK_DECLS([CPU_ALLOC], [], [], [[
 #include <sched.h>
 ]])

--- a/meson.build
+++ b/meson.build
@@ -515,6 +515,10 @@ conf.set('HAVE_ENVIRON_DECL', have ? 1 : false)
 have = cc.has_header_symbol('signal.h', 'sighandler_t', args : '-D_GNU_SOURCE')
 conf.set('HAVE_SIGHANDLER_T', have ? 1 : false)
 
+# checking sigset_t defined in the kernel header
+have = cc.has_header_symbol('asm/signal.h', 'sigset_t', args : '-D_GNU_SOURCE')
+conf.set('HAVE_SIGSET_T', have ? 1 : false)
+
 have = cc.has_function('strsignal')
 conf.set10('HAVE_STRSIGNAL_DECL', have)
 

--- a/tests/helpers/test_mkfds.c
+++ b/tests/helpers/test_mkfds.c
@@ -4536,12 +4536,14 @@ static struct multiplexer multiplexers [] = {
 		.fn = wait_event_poll,
 	},
 #endif
+#ifdef HAVE_SIGSET_T
 #ifdef __NR_ppoll
 	{
 		.name = "ppoll",
 		.fn = wait_event_ppoll,
 	},
 #endif
+#endif	/* HAVE_SIGSET_T */
 };
 
 static struct multiplexer *lookup_multiplexer(const char *name)

--- a/tests/helpers/test_mkfds_ppoll.c
+++ b/tests/helpers/test_mkfds_ppoll.c
@@ -35,6 +35,8 @@
  * This file is for defining the poll multiplexer only with <asm/poll.h>.
  *
  */
+#ifdef HAVE_SIGSET_T		/* defined in config.h */
+
 #include "test_mkfds.h"
 
 #include <string.h>		/* memset */
@@ -77,3 +79,5 @@ DEFUN_WAIT_EVENT_POLL(ppoll,
 		      clear_sigset(&sigset);,
 		      syscall(__NR_ppoll, pfds, n, NULL, &sigset, sizeof(sigset)))
 #endif
+
+#endif	/* HAVE_SIGSET_T */

